labs/b8.md

```diff
@@ -0,0 +1,84 @@
+
+![alt](https://cdn-images-1.medium.com/max/1280/1*313akmm3vKOhK0bVNMxtSQ.png =400x250)
+
+To try it out:
+`gpg --sign [FILE]` to sign `[FILE]` with your private key. `gpg --verify [FILE].gpg` to verify that the file was signed by one of public keys on your keyring. 
+
+## Hashing (Checksums)
+Hash functions deterministically map arbitrary-length data to a fixed-length hash (in other words, a fixed number of bytes). As a result, the latter can serve as a summary of the former if the former exceeds the latter in terms of byte length. For instance, if we download a 1GB file and want to verify its integrity, instead of re-downloading the entire file again, we can simply compute a 20B hash of the file on our end and compare it to the 20B hash of source.
+
+To try it out:
+1. `sha1sum [FILE]` to get the SHA1 hash of `[FILE]`.
+2. `md5sum [FILE]` to get the MD5 hash of `[FILE]`.
+
+There are [many hash functions](https://en.wikipedia.org/wiki/List_of_hash_functions), some of which are cryptographic hash functions and others not. Crytographic hash functions primarily differ from non-cryptographic hash functions in that they are difficult to forge. For instance, if someone has a hash on hand, it is difficult for them to create a pre-hash file that maps to the same hash. If you are interested in all of the properties of a cryptographic hash function, [read here](https://en.wikipedia.org/wiki/Cryptographic_hash_function). In particular, SHA1 and MD5 have been proven to no longer be cryptographically secure and are only used for checksums to ensure data integrity.
+
+
+## File Security
+The UNIX permission model has 3 components: permissions given to the file's (1) owning user, (2) owning group and (3) others/everyone else. Permissions themselves have 3 subcomponents: (1) read, (2) write and (3) execute, which are self-explanatory in respectively enforcing the ability to read, write or execute a file.
+
+![alt](https://www.comentum.com/images/permissions.jpg)
+
+To try it out:
+1. `ls -l` shows all the permissions of the current directory's files in the leftmost column.
+2. `chown [NEWUSER]:[NEWGROUP] [FILE]` to change `[FILE]`'s  user and group ownerships respectively to `[NEWUSER]` and `[NEWGROUP]`.
+3. `chmod [PERMISSIONS] [FILE]` to set `[FILE]` with specified `[PERMISSIONS]`.
+
+## Lab Checkoff
+For the lab activities: `git clone https://github.com/0xcf/decal-labs.git`. You'll be working in `b8/`.
+Here is the [checkoff form](https://docs.google.com/forms/d/e/1FAIpQLSdMDLNgyiIlXPz-DMig56Z10OAj0DU6nQOt5Ase4Y8dxoLPtw/viewform).
+
+### Encryption and Decryption
+1. Decrypt `b8/file1.txt.gpg` with the password `ocfdecal` (*for real-life purposes, never store passwords in plaintext*). What are the decrypted contents of `b8/file1.txt.gpg`?
+2. How does one import a key?
+3. How does one export a key to a file? (Hint: use `--armor`)
```

Comment:
this is not actually correct. `--export` will export a key in binary format, whereas `--armor` will ASCII-encode the binary key so it can be sent easily in text form.

Reply:
The --armor hint was misworded, I meant for them to use --armor as an additional flag to get a human-readable key and not as the way to export a key.